CONTRIBUTING.md

```diff
@@ -1,59 +1,604 @@
+
+The format can be described more formally as follows:
+
+```text
+<short title for what changed>
+<BLANK LINE>
+<why this change was made and what changed>
+<BLANK LINE>
+<any interesting details>
+<footer>
+```
+
+The first line is the subject and should be no longer than 70 characters, the
+second line is always blank, and other lines should be wrapped at a max of 80 characters.
+This allows the message to be easier to read on GitHub as well as in various git tools.
+
+There is a template recommend for use [here](https://gist.github.com/yitsushi/656e68c7db141743e81b7dcd23362f1a).
+
+# How the Maintainers process contributions
+
+## Prioritizing issues
+
+The core team regularly processes incoming issues. There may be some delay over holiday periods.
+
+Every issue will be assigned a `priority/<x>` label. The levels of priorities are:
+
+- [`critical`](https://github.com/weaveworks/flintlock/labels/priority%2Fcritical): These are time sensitive issues which should be picked up immediately.
+  If an issue labelled `critical` is not assigned or being actively worked on,
+  someone is expected to drop what they're doing immediately to work on it.
+  This usually means the core team, but community members are welcome to claim
+  issues at any priority level if they get there first. _However, given the pressing
+  timeframe, should a non-core contributor request to be assigned to a `critical` issue,
+  they will be paired with a core team-member to manage the tracking, communication and release of any fix
+  as well as to assume responsibility of all progess._
+
+- [`important-soon`](https://github.com/weaveworks/flintlock/labels/priority%2Fimportant-soon): Must be assigned as soon as capacity becomes available.
+  Ideally something should be delivered in time for the next release.
+
+- [`important-longterm`](https://github.com/weaveworks/flintlock/labels/priority%2Fimportant-longterm): Important over the long term, but may not be currently
+  staffed and/or may require multiple releases to complete.
+
+- [`backlog`](https://github.com/weaveworks/flintlock/labels/priority%2Fbacklog): There appears to be general agreement that this would be good to have,
+  but we may not have anyone available to work on it right now or in the immediate future.
+  PRs are still very welcome, although it might take a while to get them reviewed if
+  reviewers are fully occupied with higher priority issues, for example immediately before a release.
+
+- [`adr-required`](https://github.com/weaveworks/flintlock/labels/adr-required):
+  Indicates that the issue or PR contains a decision that needs to be documented in a [ADR](#adrs-architectural-decision-records) _before_
+  it can be worked on.
+
+- [`needs-investigation`](https://github.com/weaveworks/flintlock/labels/needs-investigation):  There is currently insufficient information to either categorize properly,
```

Comment:
This is `priority/awaiting-more-evidence`

Reply:
i am not a huge fan of this label as really it makes zero sense: how has it been prioritised if it still needs more evidence?

Reply:
The k8s projects also use the `needs-triage` and `triage-accepted` labels. I'm guessing the thinking is someone has read the issue and accepts it might potentially be a bug/feature but wants more information but from a priority point its the bottom of the pile.

Reply:
hmrm it's a pretty terrible label. in many situations you could end up with a `kind/bug` which is (eg) `priority/important-soon` and still needs more info (from the opener etc) to fix. Assigning `priority/awaiting-more-evidence` is confusing and misleading if it bumps it down the prio list. having just `awaiting-evidence` or `needs-info` alongside the other two is a more intuitive marker

Reply:
I agree and we don't need to follow to the letter.

The interesting part will be if we adopt Prow to manage issues/prs etc.